File: LeetCode/268.missing-number.py
# @lc code=start
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxNumber = 0
        m = {}

        for num in nums:
            maxNumber = max(maxNumber, num)
            m[num] = True

        for i in range(len(nums) + 1):
            if i not in m:
                return i
        return 1

File: LeetCode/268.missing-number.java

/*
 * @lc app=leetcode id=268 lang=java
 *
 * [268] Missing Number
 */
import java.lang.reflect.Array;
import java.util.*;

// @lc code=start

class Solution {
    public int missingNumber(int[] nums) {

        // Sorting the array
        Arrays.sort(nums);

        int missing = 0;

        for (int i = 0; i <= nums.length; i++) {
            // Check if the current number is equal to the index
            if (i >= nums.length || nums[i] != i) {
                missing = i;
                break;
            }
        }

        return missing;
    }
}
// @lc code=end
